lib/Admin_Handler.cpp

```
@@ -322,6 +342,38 @@ bool FlushCommandWrapper(S* sess, const string& modname, char *query_no_space, i
 	return false;
 }
 
+std::tuple<bool, enum SERVER_TYPE, time_t>&& parse_command_purge_query_digests(char *query, int query_len) {
+	auto ret = new std::tuple<bool, enum SERVER_TYPE, time_t>(false, SERVER_TYPE_MYSQL, 0);
+
+	const char *prefix = match_command_prefix(CMD_PREFIX_PURGE_QUERY_DIGESTS, query, query_len);
+	if (prefix) {
+		// match_found
+		std::get<0>(*ret) = true;
+
+		// server_type
+		if (strstr(prefix, "_pgsql_") != nullptr) {
+			std::get<1>(*ret) = SERVER_TYPE_PGSQL;
+		}
+
+		// parse timestamp
+		char *timestamp = strdup(query + strlen(prefix));
```

The timestamp value is not being validated. For instance:
`PURGE TABLE stats.stats_mysql_query_digest TO 123abc;`
The "abc" part is currently ignored by `atoi` (123abc -> 123), but instead of ignoring it, we should return an error to the client indicating that an invalid timestamp value was provided. This will ensure that ProxySQL behaves as expected and avoids any potential unknown issues.

---

Addressed. Now, the system throws an error - `Invalid timestamp`.